src/twisted/protocols/haproxy/test/test_wrapper.py

```diff
@@ -65,104 +68,117 @@ def test_invalidPartialHeaderDisconnects(self):
         proto.dataReceived(b"2.2.2.2 8080\r\n")
         self.assertFalse(transport.connected)
 
-    def test_validIPv4HeaderResolves_getPeerHost(self):
+    def test_preDataReceived_getPeerHost(self) -> None:
+        factory = HAProxyWrappingFactory(Factory.forProtocol(StaticProtocol))
+        proto = factory.buildProtocol(
+            address.IPv4Address("TCP", "127.0.0.1", 8080),
+        )
+        transport = StringTransportWithDisconnection(
+            hostAddress=mock.sentinel.host_address,
+            peerAddress=mock.sentinel.peer_address,
+        )
+        proto.makeConnection(transport)
+        self.assertEqual(proto.getHost(), mock.sentinel.host_address)
+        self.assertEqual(proto.getPeer(), mock.sentinel.peer_address)
```

Comment:
Minor comment... as I am a bit allergic to the mock library :)
do we need a sentinel here?
I feel that it makes the test harder to read.

```suggestion
        transport = StringTransportWithDisconnection(
            hostAddress=address.IPv4Address("TCP", "10.0.0.1", 8080),
            peerAddress=address.IPv4Address("TCP", "1.2.3.4", 1234),
        )
        proto.makeConnection(transport)
        self.assertEqual(proto.getHost(), address.IPv4Address("TCP", "10.0.0.1", 8080))
        self.assertEqual(proto.getPeer(), address.IPv4Address("TCP", "1.2.3.4", 1234))
```

Reply:
mock.sentinel is one of the nicer parts of mock imho. It makes it clear that the exact same object was passed around otherwise unobserved

Reply:
ok. no problem.  we can keep it
I guess that this is similar to `host_address = object()` but you don't need to keep the extra reference as mock.sentinel will do that for you.